hil/ext/switches/brocade.py

```diff
@@ -298,8 +299,8 @@ def _construct_tag(name):
 
     def _make_request(self, method, url, data=None,
                       acceptable_error_codes=()):
-        r = requests.request(method, url, data=data, auth=self._auth)
-        if r.status_code >= 400 and \
-           r.status_code not in acceptable_error_codes:
+        try:
+            r = requests.request(method, url, data=data, auth=self._auth)
+        except SwitchError:
```

Comment:
this won't work because `request` won't raise a switcherror. 
what we should do is
 ```python
if r.status_code >= 400 and r.status_code blah blah:
    logger.error('Bad request')
    raise SwitchError('BadRequest')
```
we raise the error here, and the networking daemon will catch it and handle it.